datacube/model/__init__.py

```diff
@@ -840,3 +927,183 @@ def metadata_from_doc(doc: Mapping[str, Any]) -> MetadataType:
     from .fields import get_dataset_fields
     MetadataType.validate(doc)  # type: ignore
     return MetadataType(doc, get_dataset_fields(doc))
+
+
+class ExtraDimensions:
+    """
+    Definition for the additional dimensions between (t) and (y, x)
+
+    It allows the creation of a subsetted ExtraDimensions that contains slicing information relative to
+    the original dimension coordinates.
+    """
+
+    def __init__(self, extra_dim: Dict[str, Any]):
+        """Init function
+
+        :param extra_dim: Dimension definition dict, typically retrieved from the product definition's
+            `extra_dimensions` field.
+        """
+        import xarray
+
+        # Dict of information about each dimension
+        self._dims = extra_dim
+        # Dimension slices that results in this ExtraDimensions object
+        self._dim_slice = {
+            name: (0, len(dim['values'])) for name, dim in extra_dim.items()
+        }
+        # Coordinate information
+        self._coords = {
+            name: xarray.DataArray(
+                data=dim['values'],
+                coords={name: dim['values']},
+                dims=(name,),
+                name=name,
+            ).astype(dim['dtype'])
+            for name, dim in extra_dim.items()
+        }
+
+    def has_empty_dim(self) -> bool:
+        """Return True if ExtraDimensions has an empty dimension, otherwise False.
+
+        :return: A boolean if ExtraDimensions has an empty dimension, otherwise False.
+        """
+        for value in self._coords.values():
+            if value.shape[0] == 0:
+                return True
+        return False
+
+    def __getitem__(self, dim_slices: Dict[str, Union[float, Tuple[float, float]]]) -> "ExtraDimensions":
+        """Return a ExtraDimensions subsetted by dim_slices
+
+        :param dim_slices: Optional cache to re-use geoboxes instead of creating new one each time
```

Comment:
is this document comment valid or a cut and paste?

Reply:
Was a cut and paste, will fix this.

Reply:
Done, Please let me know if there are any other required changes.